src/manifold/src/boolean_result.cpp

```diff
@@ -446,9 +444,10 @@ struct CreateBarycentric {
     const BaryRef oldRef = halfedgeRef.PQ == 0 ? triBaryP[tri] : triBaryQ[tri];
 
     faceRef[halfedgeR.face] = oldRef;
-    if (halfedgeRef.PQ == 0) {
-      // Mark the meshID as coming from P
-      faceRef[halfedgeR.face].meshID |= kFromP;
+
+    if (halfedgeRef.PQ == 1) {
+      // Mark the meshID as coming from Q
+      faceRef[halfedgeR.face].meshID += offsetQ;
```

Comment:
Just wondering what is the cause of the wrong meshID using bitwise or. Is it wrong because we have too many different meshes? Why adding `meshIDCounter_` is fine instead?

Reply:
I don't think that was a problem - the error was in the `UpdateMeshIDs` function, which was hitting its assertion, and that was because it on an earlier iteration it was creating an empty set of original IDs. I decided to refactor instead of debugging it and this simplified my use of an index vector.